apps/evm/src/components/DomainNameOrEllipseAddress/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { useGetAddressDomainName } from 'clients/api';
+import { Icon, Tooltip } from 'components';
+import EllipseAddress, { type EllipseAddressProps } from 'components/EllipseAddress';
+import { useTranslation } from 'libs/translations';
+import { useChainId } from 'libs/wallet';
+import { ChainId } from 'types';
+import { cn, truncateAddress } from 'utilities';
+import type { Address } from 'viem';
+
+type DomainNameAddressProps = {
+  showProvider?: boolean;
+  showTooltip?: boolean;
+  shouldEllipseAddress?: boolean;
+} & EllipseAddressProps;
+
+export const DomainNameOrEllipseAddress: React.FC<DomainNameAddressProps> = ({
+  address,
+  className,
+  showProvider = true,
+  showTooltip = true,
+  shouldEllipseAddress = true,
+  ...ellipseAddressProps
+}) => {
+  const { t } = useTranslation();
+  const { chainId } = useChainId();
+  const { data: domainNames } = useGetAddressDomainName({
+    accountAddress: address as Address,
+  });
+
+  const chainDomainName = domainNames?.[chainId];
```

Comment:
It feels unnecessary to fetch all domain names of the address if we know in advance we're only going to use the one on the current chain.

Reply:
Due to https://github.com/VenusProtocol/venus-protocol-interface/pull/3448#discussion_r1876704105, `chainId` is always passed to the request, thus requesting only what is informed.

However this doesn't change, the response is always going to be:

```
{[address]: { [chainId]: domainName}}
```

Even if there is a single chain being requested.